subworkflows/local/contaminant_filter.nf

```diff
@@ -66,11 +76,32 @@ workflow CONTAMINANT_FILTER {
 
     trna_reads.set { cdna_reads }
 
-
     if (params.cdna) {
-        BLAT_CDNA ( Channel.value( 'cdna' ), ch_reference_hairpin, ch_cdna )
+        // Search which hairpin miRNAs are present in the cDNA data
+        BLAT_CDNA(ch_reference_hairpin, ch_cdna)
         ch_versions = ch_versions.mix(BLAT_CDNA.out.versions)
-        INDEX_CDNA ( BLAT_CDNA.out.filtered_set )
+
+        // Extract the significant hits
+        ch_program = Channel.of('BEGIN{FS="\t"}{if(\$11 < 1e-5) print \$2;}').collectFile(name:"program.txt")
```

Comment:
Should we define the ch_program channel outside of the conditional statement? To avoid having the same statement repeated across all conditional blocks

Reply:
Yes, there are many pieces of repeated code. These should be improved in the future. I will address this one here but all related improvements to this subworkflow should be done separately because it might be necessary to re-think this whole subworkflow.